lgc/patch/RegisterMetadataBuilder.cpp

```diff
@@ -856,8 +864,45 @@ void RegisterMetadataBuilder::buildPsRegisters() {
 
 // =====================================================================================================================
 // Builds register configuration for compute/task shader.
-void RegisterMetadataBuilder::buildCsRegisters() {
-  // TODO: Implement once PAL completes the new format for CS
+void RegisterMetadataBuilder::buildCsRegisters(ShaderStage shaderStage) {
+  assert(shaderStage == ShaderStageCompute || shaderStage == ShaderStageTask);
+
+  getComputeRegNode()[Util::Abi::ComputeRegisterMetadataKey::TgidXEn] = true;
+  getComputeRegNode()[Util::Abi::ComputeRegisterMetadataKey::TgidYEn] = true;
+  getComputeRegNode()[Util::Abi::ComputeRegisterMetadataKey::TgidZEn] = true;
+  getComputeRegNode()[Util::Abi::ComputeRegisterMetadataKey::TgSizeEn] = true;
```

Comment:
Do you plan to optimize this later?

Reply:
Do you mean we should set `TgidZEn` for compute shaders based on `foldWorkgroupXY`?